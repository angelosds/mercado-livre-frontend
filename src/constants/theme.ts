export const ROOT_RELATIVE_SIZE = 16
