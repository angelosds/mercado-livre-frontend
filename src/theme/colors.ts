import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  primary: '#fe6000',
  secondary: '#3483fa',
  gray900: '#333333',
  gray700: '#666666',
  gray500: '#999999',
  gray300: '#eeeeee',
  white: '#ffffff',
};
