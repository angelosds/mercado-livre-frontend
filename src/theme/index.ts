import { DefaultTheme } from 'styled-components';

import { pixelToRem } from '../utils';

export const theme: DefaultTheme = {
  colors: {
    primary500: '#fee600',

    secondary100: '#c6dcf7',
    secondary500: '#3483fa',
    secondary700: '#2968c8',

    gray900: '#333333',
    gray700: '#333333',
    gray500: '#666666',
    gray400: '#999999',
    gray300: '#bbbbbb',
    gray200: '#dddddd',
    gray100: '#eeeeee',

    border200: '#f7f7f7',

    success: '#1ed760',

    white: '#ffffff',
  },
  fontSizes: {
    xs: pixelToRem(14),
    sm: pixelToRem(16),
    md: pixelToRem(24),
    lg: pixelToRem(28),
    xl: pixelToRem(46),
  },
  outline: {
    sm: '3px'
  },
  space: {
    sm: '4px',
    md: '8px',
    lg: '16px',
    xl: '32px',
  },
  radius: {
    sm: '4px'
  },
  breakpoints: {
    mobile: '565px'
  }
};
