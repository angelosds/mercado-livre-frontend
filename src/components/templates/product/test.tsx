import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components';

import { theme } from '../../../theme';

import ProductDetailsTemplate, { Props } from '.'

const defaultProps: Props = {
  product: {
    condition: 'Novo',
    description: 'Descrição',
    free_shipping: true,
    id: 'MLA902237011',
    picture: 'https://http2.mlstatic.com/D_NQ_NP_974179-MLB43818784989_102020-W.webp',
    price: 1200.99,
    sold_quantity: 30,
    state_name: 'São Paulo',
    title: 'Nome produto',
  },
  onBuyItem: () => {}
}

describe('ProductDetailsTemplate', () => {
  const build = ({ product, onBuyItem }: Props) => {
    const { container } = render(<ThemeProvider theme={theme}><ProductDetailsTemplate product={product} onBuyItem={onBuyItem} /></ThemeProvider>)

    return { container }
  }

  it('should render the component', () => {
    const { container } = build(defaultProps)

    expect(container.firstChild).toMatchSnapshot()
  })
})
