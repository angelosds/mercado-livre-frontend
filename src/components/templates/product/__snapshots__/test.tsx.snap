// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetailsTemplate should render the component 1`] = `
<div
  class="sc-gsDKAQ bgFRru"
  data-testid="card-wrapper"
>
  <section
    class="sc-dkPtRN ciGVzC"
  >
    <div
      class="sc-hKwDye eWEWxW"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
        />
        <img
          alt="Nome produto"
          data-nimg="responsive"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fhttp2.mlstatic.com%2FD_NQ_NP_974179-MLB43818784989_102020-W.webp&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
      </span>
      <div
        class="sc-eCImPb CcoOd"
      >
        <p
          class="sc-jRQBWg cwjEuF"
          data-testid="details"
        >
          productCondition.Novo
           - 
          selled
        </p>
        <h1
          class="sc-gKclnd gYcJiM"
        >
          Nome produto
        </h1>
        <button
          class="sc-bdvvtL iMiavG"
          type="button"
        >
          buy
        </button>
        <p
          class="sc-iCfMLu FuuQL"
          data-testid="price"
        >
          $ 
          1.200
          <span
            class="sc-furwcr iyOiGj"
          >
            99
          </span>
        </p>
      </div>
    </div>
    <div
      class="sc-hKwDye eWEWxW"
    >
      <div>
        <h2
          class="sc-pVTFL iXEJci"
        >
          productDescription
        </h2>
        <p
          class="sc-jrQzAO cYTIJZ"
          data-testid="description"
        >
          Descrição
        </p>
      </div>
    </div>
  </section>
</div>
`;
